riscv64-link.c

```diff
@@ -164,7 +164,9 @@ ST_FUNC void relocate_plt(TCCState *s1)
     if (s1->plt->reloc) {
         ElfW_Rel *rel;
         p = s1->got->data;
-        for_each_elem(s1->plt->reloc, 0, rel, ElfW_Rel) {
+        for (rel = (ElfW_Rel *) s1->plt->reloc->data; \
```

Comment:
What does it need this change?

Reply:
The macro `for_each_elem` is not properly properly imported here. It's defined in a `.c` file, so I don't think importing it is a good idea unless we relocate it to a header (`tcc.h` seems a natural choice if that's the preferred route).

Reply:
Is ONESOURCE helping with it? probably right?

Reply:
It isn't, actually. I tried reverting this change with that flag and the build failed.

Reply:
aiight!